raft/src/main/java/org/apache/kafka/raft/RaftUtil.java

```diff
@@ -159,4 +161,18 @@ static boolean hasValidTopicPartition(DescribeQuorumRequestData data, TopicParti
                    data.topics().get(0).partitions().size() == 1 &&
                    data.topics().get(0).partitions().get(0).partitionIndex() == topicPartition.partition();
     }
+
+    static boolean hasValidTopicPartition(FetchSnapshotRequestData data, TopicPartition topicPartition) {
```

Comment:
How about changing this to return an `Errors`?
1. `INVALID_REQUEST` if there is more than one topic partition
2. `UNKNOWN_TOPIC_OR_PARTITION` if the topic partition doesn't match the log's name and partition
3. `NONE` otherwise

Reply:
I think this is a viable improvement, there are 11 similar methods here, so I changed them all and add a unit test for them.